### ISSUE TITLE
Bundle fails on dsn define

### ISSUE THREAD

**Issue by jfuentesb** [backup-manager/symfony#75]
mysql://user:pass@server:3306/database

And show me:
mysqldump: Got error: 1049: "Unknown database '/database'" when selecting the database. I think that this take the '/' too.

**Comment by frvaillant** [comment]
I got the same issue. There is also a problem when database password contains "@"

**Comment by lhpalacio** [comment]
@jfuentesb and @frvaillant nice catch! for now is better to avoid DSN config.

A workaround is set database connection config like this:
```
bm_backup_manager:
    database:
        development:
            type: mysql
            host: localhost
            port: 3306
            user: root
            pass: pass@word
            database: test
```

@Nyholm A `parse_url` function is not enough to parser the DSN config?

**Comment by Nyholm** [comment]
Thank you for the ping. 

@jfuentesb I've added a PR to fix this this issue. 

@frvaillant That is correct. Using `@` in the password is not valid according to [RFC 3986](https://tools.ietf.org/html/rfc3986). You must `urlencode` it and write it as "%40". 

@lhpalacio That is a good question. It actually isn't. 
If we were to use `parse_url`, then we need to `urldecode` authentication strings. We also needed to implement a custom error handler for invalid strings. 

Aslo, I think `parse_url` has some issues with some non English alphabets.

**Comment by frvaillant** [comment]
I didn't know for @. Thanks a lot.

**Comment by frvaillant** [comment]
@Nyholm I tried both
pass: %40MYpassWord
pass: "%40MyPassWord"
pass: '%40MyPassWord'

In each case, i get "The provided DSN is not valid. Maybe you need to url-encode the user/password?" as error

Did i make something wrong or just shoud I change my password ? Thanks for help

**Comment by Nyholm** [comment]
Hey. 

So if you have the following: 
user: fancois
password: p@ssword

The DSN (without encoding would look like: `mysql://fancois:p@ssword@server:3306/database`
But that is not valid, So you need to run `urlencode()`on your password. The new DSN looks like: `mysql://fancois:p%40ssword@server:3306/database`.

Generally, when you are building your DSN, you should always urlencode password and username. Ie: 

```php
echo sprintf('mysql://%s:%s@server:3306/database', urlencode('fancois'), urlencode('p@ssword'));
```

https://3v4l.org/Vm3si

I hope this makes sense.

**Comment by jfuentesb** [comment]
@Nyholm  thanks!

**Comment by Nyholm** [comment]
You are welcome. I've released 3.1.1

**Comment by frvaillant** [comment]
Great job @Nyholm . Thanks. 
I updated Backup Manager on my project, and it works perfectly (had to create a new mysql user to resolve my @ problem more simply)
